### DISCUSSION TITLE
Is there a reason to use 'conda run' instead of 'conda activate' in the Dockerfile?

### DISCUSSION THREAD

**Discussion by MartinMeilinger** [QUARK-framework/QUARK#41] [General]
In the current version of the Dockerfile (on the dockerSupport branch), a conda environment is created using the .yml file:
`RUN conda env create -f environment.yml`

After this, the conda environment is used with `conda run` instead of using `conda activate`.
This means that `conda run` needs to be prepended to every command. In case of using `conda activate`, you could just write the command without needing to prepend anything.
(cf [conda docs](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#managing-environments))

Is there a reason not to activate the environment?

**Comment by MartinMeilinger** [comment]
Also, would it be possible to rebase + force push the dockerSupport to the up-to-date commit on the main branch (or rebase and push on a new branch)? That way, my typo fixes and stuff, including putting '/.idea/' into '.gitignore', would be present. It's not a huge issue, but would be nice.

**Reply by philross** [reply]
I just rebased the branch, so your recent changes should be present.

**Reply by MartinMeilinger** [reply]
Thank you! :D

**Comment by philross** [comment]
> In the current version of the Dockerfile (on the dockerSupport branch), a conda environment is created using the .yml file: `RUN conda env create -f environment.yml`
> 
> After this, the conda environment is used with `conda run` instead of using `conda activate`. This means that `conda run` needs to be prepended to every command. In case of using `conda activate`, you could just write the command without needing to prepend anything. (cf [conda docs](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#managing-environments))
> 
> Is there a reason not to activate the environment?

There is no reason not to activate it, so I guess activating the environment is the better option.

**Comment by MartinMeilinger** [comment]
So, I tried to activate conda for the Docker container, but failed to do so.
In short, to be able to use `conda activate`, conda needs to initialize some things in a way that conflicts with how Docker images are built. The problem is explained in more detail here: [https://pythonspeed.com/articles/activate-conda-dockerfile/](https://pythonspeed.com/articles/activate-conda-dockerfile/)
The solution proposed in this article is exactly what the Dockerfile does in its current form. Therefore, I'll leave it at that.
